File: mha/track_mha.m
function track_mha(srcdir, srcpattern, lookahead, load_data, destmat)
%TRACK_MHA Track eddies using Multiple Hypothesis Assignment
% srdir:      Directory in which to find saved eddy frames
% srcpattern: Filename pattern used for eddies (ex. anticyc_ for
%             anticyc_19921014.mat)
% lookahead:  Boolean value to run the scan with lookahead enabled [0 or 1]
% load data:  Path to pre-generated data. Use 0 to load no data.
% destmat:    Path to saved mat-file. (should end in .mat)
    if isa(load_data, 'numeric')
        load_data = '0';
    end
    
    mha_path = mfilename('fullpath');
    sep = strfind(mha_path, filesep());
    mha_path = [mha_path(1:sep(end)) 'track_mha.py'];
    
    cmd = sprintf('python "%s" "%s" "%s" %d "%s" "%s"', mha_path, ...
        srcdir, srcpattern, lookahead, load_data, destmat);
    status = system(cmd, '-echo');
    if status
        error('Failed to run MHA');
    end
end

File: mha/track_mha.py
#!/usr/bin/env python

import mht
import os
import sys

def usage():
	print 'Usage:  {} [srcdir] [srcpattern] [lookahead] [load data] [destmat]'.format(
		os.path.basename(__file__))
	print '    srdir:      Directory in which to find saved eddy frames'
	print '    srcpattern: Filename pattern used for eddies (ex. anticyc_ for'
	print '                anticyc_19921014.mat)'
	print '    lookahead:  Boolean value to run the scan with lookahead enabled [0 or 1]'
	print '    load data:  Path to pre-generated data. Use 0 to load no data.'
	print '    destmat:    Path to save mat-file(s). (should NOT end in .mat)'

if len(sys.argv) != 6:
	usage()
	sys.exit(1)

srcdir = sys.argv[1]
srcpattern = sys.argv[2]
do_look = sys.argv[3].lower() in ['1', 'yes', 'true']
if sys.argv[4].lower() in ['0', 'no', 'false']:
	load_path = None
else:
	load_path = sys.argv[4]
destmat = sys.argv[5]

if load_path is None:
	data = None
else:
	print 'Loading old data...'
	data = mht.load_tracks(load_path)
print 'Tracking...'
roots = mht.build_mht(mht.list_eddies(srcdir, srcpattern), do_lookahead=do_look, prev_data=data)
print 'Saving...'
mht.write_tracks(roots, destmat, mht.list_dates(srcdir, srcpattern))
